gtsfm/averaging/translation/averaging_1dsfm.py

```diff
@@ -48,13 +62,42 @@ def __init__(self, robust_measurement_noise: bool = True) -> None:
         self._max_1dsfm_projection_directions = MAX_PROJECTION_DIRECTIONS
         self._outlier_weight_threshold = OUTLIER_WEIGHT_THRESHOLD
 
+    def __sample_projection_directions(
+        self, w_i2Ui1_measurements: BinaryMeasurementsUnit3, projection_sampling_method: ProjectionSamplingMethod
+    ) -> List[Unit3]:
+        """Samples projection directions for 1DSfM based on the provided sampling method.
+
+        Args:
+            w_i2Ui1_measurements: Unit translation measurements which are input to 1DSfM.
+            projection_sampling_method: ProjectionSamplingMethod to be used for sampling directions.
+
+        Returns:
+            List of sampled Unit3 projection directions.
+        """
+        num_measurements = len(w_i2Ui1_measurements)
+
+        if projection_sampling_method == self.ProjectionSamplingMethod.SAMPLE_INPUT_MEASUREMENTS:
+            num_samples = min(num_measurements, self._max_1dsfm_projection_directions)
+            sampled_indices = np.random.choice(w_i2Ui1_measurements, num_samples, replace=False)
+            return [w_i2Ui1_measurements[idx].measured() for idx in sampled_indices]
+
+        elif projection_sampling_method == self.ProjectionSamplingMethod.SAMPLE_WITH_INPUT_DENSITY:
+            return _sample_kde_directions(w_i2Ui1_measurements, num_samples=self._max_1dsfm_projection_directions)
+
+        elif projection_sampling_method == self.ProjectionSamplingMethod.SAMPLE_WITH_UNIFORM_DENSITY:
+            return _sample_random_directions(num_samples=self._max_1dsfm_projection_directions)
```

Comment:
Add these options to the config?

Reply:
I would not want to, because for that we will need the parameter in the constructor of the base class, and this is very implementation specific (should not go in the base class). 
Unless there is a way in the config to provide options for derived classes.

Reply:
You could just make it an attribute of the derived class, similar to how you did it for the [`CycleConsistentRotationViewGraphEstimator`](https://github.com/borglab/gtsfm/blob/3eaa76e6439f9a4f6e8565bb0c42b8ce92fcdb13/gtsfm/view_graph_estimator/cycle_consistent_rotation_estimator.py#L47).

Reply:
yeah i think it might be nice to put that in the constructor as default

Reply:
updated